src/main/kotlin/com/openlattice/apps/services/AppService.kt

```diff
@@ -435,4 +455,12 @@ class AppService(
         }
     }
 
+    private fun ensureAppRolesExist(app: App, roleIds: Set<UUID>) {
+        val missingRoleIds = Sets.difference(roleIds, app.appRoles.map { it.id }.toSet())
```

Comment:
roleIds  - app.appRoles, the view created by Sets.difference is expensive and you have to create another set by calling .toSet() anyway.